src/main/scala/xiangshan/frontend/icache/ICacheMainPipe.scala

```diff
@@ -421,10 +422,10 @@ class ICacheMainPipe(implicit p: Parameters) extends ICacheModule
   /** s2 control */
   val s2_fetch_finish = Wire(Bool())
 
-  val s2_valid          = generatePipeControl(lastFire = s1_fire, thisFire = s2_fire, thisFlush = false.B, lastFlush = false.B)
```

Comment:
For the S2 pipeline stage, a direct flush may not be appropriate because there could be ongoing processing of miss.